pkg/storage/azure/azureclient/azureclient.go

```diff
@@ -163,20 +163,27 @@ func (c *Client) GetVNetByTag(ctx context.Context, resourceGroupName, tagKey str
 		if err != nil {
 			return armnetwork.VirtualNetwork{}, err
 		}
+		// there are a lot of nested loops here, I know.
+		// even if there are a lot of vnets to loop through,
+		// we don't expect there to be a lot of entries in tagFilter,
+		// so the inner loops will be fairly self-contained, at least
+		// for the use cases in this operator.
 		for _, vnet := range page.Value {
-			tag, ok := vnet.Tags[tagKey]
-			if !ok {
-				continue
-			}
-			for _, tagValue := range tagValues {
-				if *tag == tagValue {
-					return *vnet, nil
+			for tagKey, tagValues := range tagFilter {
+				tag, ok := vnet.Tags[tagKey]
+				if !ok {
+					continue
+				}
+				for _, tagValue := range tagValues {
+					if *tag == tagValue {
+						return *vnet, nil
+					}
 				}
 			}
```

Comment:
Not sure if it's worth it, but this helps a bit with the reading:

```suggestion
             if vnetHasAnyTag(vnet, tagFilter) {
                    return *vnet, nil
             }
```
```
func vnetHasAnyTag(vnet *armnetwork.VirtualNetwork, tagFilter map[string][]string) bool {
		for tagKey, tagValues := range tagFilter {
				tag, ok := vnet.Tags[tagKey]
				if !ok {
					continue
				}
				for _, tagValue := range tagValues {
					if *tag == tagValue {
						return true
					}
				}
		}
        return false
}
```

Reply:
Good point, thanks! I'm on it.

Reply:
done